qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -1,26 +1,114 @@
 #!/bin/sh -ex
 #
-# rbd_mirror_journal.sh - test rbd-mirror daemon in journal-based mirroring mode
+# rbd_mirror.sh - test rbd-mirror daemon in snapshot and journal-based mirroring mode
+#
+# Usage:
+#   rbd_mirror.sh <options>
+#
+# options:
+#   -h or --help    show this message" >&2
+#   -m or --mode    mirroring mode (snapshot or journal)
+#
+# example:
+#   rbd_mirror.sh --mode snapshot
 #
 # The scripts starts two ("local" and "remote") clusters using mstart.sh script,
 # creates a temporary directory, used for cluster configs, daemon logs, admin
 # socket, temporary files, and launches rbd-mirror daemon.
 #
 
-. $(dirname $0)/rbd_mirror_helpers.sh
+PROGNAME=$(basename $0)
+JOURNAL_MODE=journal
+SNAPSHOT_MODE=snapshot
+
+function usage() {
+  cat <<EOF
+Usage:
+  ${PROGNAME} <options>
+
+options:
+  -h or --help    show this message" >&2
+  -m or --mode    mirroring mode (${SNAPSHOT_MODE} or ${JOURNAL_MODE})
+
+example:
+  ${PROGNAME} --mode snapshot
+EOF
+}
+
+# parse command line arguments
+function parseargs() {
+  # Short option flags
+  SHORT_OPTS=""
+  SHORT_OPTS="${SHORT_OPTS},h"
+  SHORT_OPTS="${SHORT_OPTS},m:"
+
+  # Short option flags
+  LONG_OPTS=""
+  LONG_OPTS="${LONG_OPTS},help"
+  LONG_OPTS="${LONG_OPTS},mode:"
+
+  TEMP=$(getopt --name "${PROGNAME}" \
+    --options "${SHORT_OPTS}" \
+    --longoptions "${LONG_OPTS}" \
+    -- "$@")
+	eval set -- "$TEMP"
+  while true; do
+    case "$1" in
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      -m|--mode)
+        MODE=$(echo "$2" | awk '{print tolower($0)}')
+        if [ "${MODE}" != "${SNAPSHOT_MODE}" ] && [ "${MODE}" != "${JOURNAL_MODE}" ]; then
+          echo "unknown mode: ${2}"
+          usage
+          exit 100
+        fi
+        shift 2
+        ;;
+      --)
+        shift
+        break
+        ;;
+      *)
+        echo "unexpected argument $1" 1>&2
+        exit 100
+        ;;
+    esac
+  done
+
+  # --mode is a mandatory argument to program
+  if [ -z ${MODE} ]; then
+    echo "hint: '-m | --mode' is mandatory argument"
+    usage
+    exit 100
+  fi
+}
+
+parseargs "$@"
```

Comment:
I wonder if a screenful of usage and parseargs code is warranted for a workunit script.  Did you consider taking the mode as a positional argument (i.e. without `-m/--mode`) or as an environment variable?  The latter would also be much easier to hook up in teuthology.

Reply:
Sure.